pkg/cloudprovider/aws/fleet/launchtemplate.go

```diff
@@ -55,25 +57,42 @@ type LaunchTemplateProvider struct {
 	clientSet               *kubernetes.Clientset
 }
 
-func (p *LaunchTemplateProvider) Get(ctx context.Context, cluster *v1alpha1.ClusterSpec) (*ec2.LaunchTemplate, error) {
-	if launchTemplate, ok := p.launchTemplateCache.Get(cluster.Name); ok {
+// Translate architecture into AWS-recognized name for Bottlerocket
+// AMI.
+func normalizeArchitecture(architecture v1alpha1.Architecture) string {
+	switch architecture {
+	case v1alpha1.ArchitectureAmd64:
+		return "x86_64"
+	default:
+		return string(architecture)
+	}
+}
+
+func launchTemplateName(clusterName string, arch string) string {
+	return fmt.Sprintf(launchTemplateNameFormat, clusterName, arch)
+}
+
+func (p *LaunchTemplateProvider) Get(ctx context.Context, cluster *v1alpha1.ClusterSpec, constraints *cloudprovider.Constraints) (*ec2.LaunchTemplate, error) {
+	arch := normalizeArchitecture(*constraints.Architecture)
```

Comment:
Grab this from constraints.Architecture. It should always be populated correctly, now

Reply:
done

Reply:
The defaulting logic is already handled for you, you can just cut this normalizeArchitecture function entirely.

Reply:
Ah gotcha.